docs/paper/dev/api/event-api/chat-event.md

```diff
@@ -0,0 +1,154 @@
+---
+slug: /dev/chat-events
+description: How to properly use the new AsyncChatEvent 
+---
+
+# Chat Events
+The chat event has evolved a couple of times over the years. 
+This guide will explain how to properly use the new `AsyncChatEvent` and its `Renderer`.
+
+## Understanding the Renderer
+
+Before we can start using the new chat event, we need to understand how the new renderer works.
+The renderer is Paper's way of allowing plugins to modify the chat message before it is sent to the player.
+This is done by using the `ChatRenderer` interface with its `render` method.
+
+```java title="ChatRenderer#render"
+public @NotNull Component render(@NotNull Player source, @NotNull Component sourceDisplayName, @NotNull Component message, @NotNull Audience viewer) {
+    // ...
+}
+```
+
+- The `render` method is called when a chat message is sent to the player.
+- The `source` parameter is the player that sent the message.
+- The `sourceDisplayName` parameter is the display name of the player that sent the message.
+- The `message` parameter is the message that was sent.
+- The `viewer` parameter is the player that is receiving the message.
+
+Now that we understand how the renderer works, we can start using it.
```

Comment:
I think before this happens, it needs to be made clear that the render method is something that plugins implement, they generally do not call it themselves. And they don't supply each of those arguments, its given to them when the render method is called.

Reply:
I do that in the next section.